pkg/splunk/enterprise/util.go

```diff
@@ -630,3 +617,99 @@ func markAppsStatusToComplete(appSrcDeplymentStatus map[string]enterprisev1.AppS
 
 	return err
 }
+
+// setupAppInitContainers creates the necessary shared volume and init containers to download all
+// app packages in the appSources configured and make them locally available to the Splunk instance.
+func setupAppInitContainers(client splcommon.ControllerClient, cr splcommon.MetaObject, podTemplateSpec *corev1.PodTemplateSpec, appFrameworkConfig *enterprisev1.AppFrameworkSpec) {
+	// Create shared volume and init containers for App Framework
+	if len(appFrameworkConfig.AppSources) > 0 {
+		// Create volume to shared between init and Splunk container to contain downloaded apps
+		emptyVolumeSource := corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		}
+
+		initVol := corev1.Volume{
+			Name:         appVolumeMntName,
+			VolumeSource: emptyVolumeSource,
+		}
+
+		podTemplateSpec.Spec.Volumes = append(podTemplateSpec.Spec.Volumes, initVol)
+
+		// Add init apps mount to Splunk container
+		initVolumeSpec := corev1.VolumeMount{
+			Name:      appVolumeMntName,
+			MountPath: appBktMnt,
+		}
+
+		// This assumes the Splunk instance container is Containers[0], which I *believe* is valid
+		podTemplateSpec.Spec.Containers[0].VolumeMounts = append(podTemplateSpec.Spec.Containers[0].VolumeMounts, initVolumeSpec)
+
+		// Add app framework init containers per app source and attach the init volume
+		for i, appSrc := range appFrameworkConfig.AppSources {
+			// Get volume info from appSrc
+			volSpecPos, err := checkIfVolumeExists(appFrameworkConfig.VolList, appSrc.VolName)
+			if err != nil {
+				// Invalid appFramework config.  This shouldn't happen
```

Comment:
Should we add an error log here before continuing? Same for the other continue in the code block

Reply:
Added the debug messages